src/cfgparse.c

```diff
@@ -156,10 +173,12 @@ int str2listener(char *str, struct proxy *curproxy, struct bind_conf *bind_conf,
 			*next++ = 0;
 		}
 
-		ss2 = str2sa_range(str, NULL, &port, &end, &fd, &proto, NULL, err,
+	
+		ss2 = str2sa_range(tcp4_prefix(str), NULL, &port, &end, &fd, &proto, NULL, err,
```

Comment:
can you remove `tcp4_prefix()`: I'm not sure to understand what you want to do (and will cause memleak)